tests/core/api/test_userportalsync_create.py

```diff
@@ -0,0 +1,236 @@
+from fastapi.testclient import TestClient
+from sqlalchemy.orm import sessionmaker
+from pyispyb.app.extensions.database.middleware import Database
+from pyispyb.app.main import app
+from pyispyb.config import settings
+from pyispyb.app.extensions.database.session import engine
+from pyispyb.core.modules.proposals import get_proposals
+from pyispyb.core.modules.persons import get_persons
+from pyispyb.core.modules.laboratories import get_laboratories
+from pyispyb.core.modules.proteins import get_proteins
+from pyispyb.core.modules.sessions import get_sessions
+from pyispyb.core.modules.labcontacts import get_labcontacts
+from tests.core.api.utils.permissions import mock_permissions
+from tests.core.api.data.userportalsync_create import (
+    test_data_proposal_userportalsync_create,
+)
+
+
+TestingSessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)
+session = TestingSessionLocal()
+db = Database()
+db.set_session(session)
+
+
+client = TestClient(app)
```

Comment:
Theres a fixture to provide this too https://github.com/ispyb/py-ispyb/blob/master/tests/conftest.py#L17
Just add `client` as a parameter to your test and pytest will autoload the fixture